content/en/posts/js/introduction-a-reactjs/index.md

```diff
@@ -0,0 +1,176 @@
+
+Basically, Angular is technically really interesting (but don't get me started
+on dirty checking)
+
+### Ember
+
+[Ember](http://emberjs.com) is a well-thought and well-developed framework.
+It integrates `data-binding` thanks to [DOM Ranges](https://developer.mozilla.org/en-US/docs/Web/API/range).
+It also has strong design patterns and contrary to what people commonly
+think, it's actually pretty straightforwad.
+Subviews are also pretty easy to use thanks to `{{outlet}}`.
+
+In another words, Ember and Angular both come bundled with real
+solutions for UI management. But they also share the same way to
+do it :
+
+- _rendering_ (creating and inserting DOM nodes) once
+- updating the bindings
+
+### React
+
+[React](http://facebook.github.io/react/) doesn't care what other people do.
+It has one simple way of going about stuff :
+holy crap, managing the state of the DOM _sucks_
+
+Okay, let's just say we call `.render()` everytime anything changes.
+That might sound _real_ dumb but there's a gotcha about it.
+
+React has a __virtual DOM__ (nothing to do with facebook's purchase of Oculus VR),
+a blazing-fast internal representation of the DOM. It comes with its own events system
+which allows React to bring the capturing phase to browsers that don't have
+`EventTarget` (looking at you, IE8).
+
+The `render` method returns objects corresponding to the internal representation
+of the virtual DOM.
+
+React classes are defined by their `state`.
+When creating a class, you specify a `getInitialState` method which
+will return the initial state (duh!).
+
+After that, the only way to modify the state is to pass which values
+of the state have changed to `this.setState` so that it can update
+the DOM.
+
+A React class takes properties when it's instanciated : they're called `props`.
+They shouldn't be confused with the `state`, its content should only be manipulated
+from _outside_ of the the class (even though it can still get default values by
+specifying a `getDefaultProps` method which will return them).
+
+The `state`, however, shall only be modified by the class's own methods.
+
+The main advantage to this is that you're always sure, thanks to the systematic
+call to `render`, that your React component will look and act like you expect it
+to.
+
+Another advantage to React is its in-house diff algorithm.
+The virtual DOM gets diffed against the previous one (which is the one you see),
+and React works out which operations it needs to execute in order to update the 
+DOM.
+
+This solves a few issues like keeping track of the caret's position in a text
+field that uses two-way data-binding : since the algorithm doesn't see any reason
+to update it, the text field doesn't get re-rendered and you keep the focus.
+In the same way, if you've got a gif looping somewhere in your page, it will keep
+playing as it normally would.
+
+React is made to work with JSX (that's JavaScript XML), a superset of JS which
```

Comment:
also maybe remove `(that's JavaScript XML)`, redundant with the following `XML syntax`